ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/StateMachine.hs

```diff
@@ -1567,6 +1572,22 @@ prop_trace trace = invalidBlockNeverValidatedAgain
     isOpened (TraceOpenEvent (OpenedDB {})) = True
     isOpened _                              = False
 
+    tentativeHeaderMonotonicity =
+        counterexample "Trap tentative headers did not improve monotonically" $
+        all isMonotonicallyIncreasing trapTentativeSelectViews
+      where
+        isMonotonicallyIncreasing as = and $ zipWith (<) as (drop 1 as)
+
+        trapTentativeSelectViews =
```

Comment:
Please add a signature here, since it's a bid unusual that a name ending in `s` actually has two layers of `[]` in its type.